src/query/service/src/procedures/systems/fuse_block.rs

```diff
@@ -0,0 +1,69 @@
+//  Copyright 2022 Datafuse Labs.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+use std::sync::Arc;
+
+use common_datablocks::DataBlock;
+use common_datavalues::DataSchema;
+use common_exception::Result;
+
+use crate::procedures::OneBlockProcedure;
+use crate::procedures::Procedure;
+use crate::procedures::ProcedureFeatures;
+use crate::sessions::QueryContext;
+use crate::sessions::TableContext;
+use crate::storages::fuse::table_functions::FuseBlock;
+use crate::storages::fuse::FuseTable;
+
+pub struct FuseBlockProcedure {}
+
+impl FuseBlockProcedure {
+    pub fn try_create() -> Result<Box<dyn Procedure>> {
+        Ok(FuseBlockProcedure {}.into_procedure())
+    }
+}
+
+#[async_trait::async_trait]
+impl OneBlockProcedure for FuseBlockProcedure {
+    fn name(&self) -> &str {
+        "FUSE_BLOCK"
+    }
+
+    fn features(&self) -> ProcedureFeatures {
+        ProcedureFeatures::default().num_arguments(3)
```

Comment:
I think we should let the user get the snapshot, we can do it with the latest snapshot if the argument is 2.
```
fuse_block('db', 'table')
```

Reply:
```sql
fuse_block('db', 'table'[,'snapshot_id'])
```